### ISSUE TITLE
Viewer.destroy() doesn't remove inline CSS on target and destroys entire target contents

### ISSUE THREAD

**Issue by julmot** [openseadragon/openseadragon#1353]
https://openseadragon.github.io/docs/OpenSeadragon.Viewer.html#destroy

When calling `Viewer.destroy()` the entire OpenSeadragon element is removed. However, `stylie="position:relative;"` is added to the target (`element` option). This relative positioning isn't removed.

Also, what this does is not only removing `div.openseadragon-container` but the entire target element. This is a problem if you want to append OpenSeadragon to an element and this works, but removing it inside this appended container doesn't work since it destroys also other elements inside the target.

**Comment by iangilman** [comment]
Good point! Thank you for reporting this. 

Would you be up for writing a patch to fix it?

**Comment by julmot** [comment]
Yeah I'm open for it, but we should discuss the solution before I'm starting with it.

From my point of view, the safest solution would be not to manipulate the provided target at all. There may be the case that position relative is already set or is set during OpenSeadragon is active, so there's no guarantee that it'd remove an inline style that is set by OpenSeadragon – it could break an application.
Another approach would be to create the required container with position relative by OpenSeadragon including `width:100%;height:100%;`. Then it'd be possible to safely remove this container too upon destroying and we wouldn't need to manipulate the actual target.

**Comment by iangilman** [comment]
If we create the container with `width: 100%`, we don't know what its positioning parent is. I think we do need to set position relative on the container we're given, but we can record what its position style is prior to doing so, so we can reset it in `destroy`. Does that sounds reasonable?

**Comment by julmot** [comment]
> we don't know what its positioning parent is

I'm not sure if I'm understanding the problem with my solution approach. Why do we need to know the positioning of the target (`element` option)? Why wouldn't it be enough to know that the container we've inserted into the target has `position:relative`?

> but we can record what its position style is prior to doing so

As mentioned there may be the case that the positioning is changed during the lifetime of OpenSeadragon, thus we have no guarantee that it's the position relative of OpenSeadragon we're destroying or the positioning of the application we're running in. But to be fair, the risk is very low.

**Comment by iangilman** [comment]
> I'm not sure if I'm understanding the problem with my solution approach. Why do we need to know the positioning of the target (element option)? Why wouldn't it be enough to know that the container we've inserted into the target has position:relative?

The goal is to match the position of the element we're given. We're going to `width: 100%` inside of it so our canvas is the same width as the element we're given. The only way that'll work is if the element we're given isn't `position: static` because if it is static, the 100% won't be relative to it. See https://css-tricks.com/absolute-positioning-inside-relative-positioning/ for more info.

I agree that there's a chance that the positioning can be changed during the lifetime of the OSD viewer, but since we have this requirement, it's certainly not a good idea for someone to do so. In practice, I don't think it comes up (considering this is how it's already been for years now). Of course if there's an even cleaner solution, I'm open to exploring it!
